static/js/containers/ProfilePage_test.js

```diff
@@ -447,4 +451,41 @@ describe("ProfilePage", function() {
       });
     }
   }
+
+  it('shows a spinner when profile get is processing', () => {
+    return renderComponent('/profile').then(([, div]) => {
+      assert.notOk(div.querySelector(".spinner"), "Found spinner but no fetch in progress");
+      helper.store.dispatch({
+        type: REQUEST_GET_USER_PROFILE,
+        payload: {
+          username: SETTINGS.username
+        }
+      });
+
+      assert(div.querySelector(".spinner"), "Unable to find spinner");
+    });
+  });
+
+  it('shows an error when profile get has errored', () => {
+    return renderComponent('/profile').then(([, div]) => {
+      assert.notOk(div.querySelector(".spinner"), "Found spinner but no fetch in progress");
+      helper.store.dispatch({
+        type: REQUEST_GET_USER_PROFILE,
+        payload: {
+          username: SETTINGS.username
+        }
+      });
+
+      assert(div.querySelector(".spinner"), "Unable to find spinner");
+    });
+  });
+
+  it('renders errors when there is an error receiving the profile', () => {
+    helper.profileGetStub.returns(Promise.reject(ERROR_RESPONSE));
+
+    const types = [RECEIVE_DASHBOARD_SUCCESS, RECEIVE_GET_USER_PROFILE_FAILURE];
+    return renderComponent("/profile", types, false).then(([, div]) => {
+      assert(div.getElementsByClassName('alert-message').length > 0, 'no alert message found');
+    });
+  });
```

Comment:
Is there any value in adding tests to check the different error cases we have for the dashboard? (error code, user error message and so on)